File: unity/Assets/Scripts/Multiplayer/Players/PlayerBanResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using UnityEngine;

namespace Michitai.Multiplayer.Players
{
    /// <summary>
    /// Response returned when a player is successfully banned.
    /// Contains details about the ban including duration and expiration.
    /// </summary>
    [Serializable]
    public class PlayerBanResponse : ApiResponse<ECommonError>
    {
        /// <summary>
        /// Confirmation message for the ban.
        /// </summary>
        public string message;

        /// <summary>
        /// The unique ID of the ban record.
        /// </summary>
        public string ban_id;

        /// <summary>
        /// The ID of the banned player.
        /// </summary>
        public int player_id;

        /// <summary>
        /// The duration of the ban (e.g., "1 hour", "1 day", "permanent").
        /// </summary>
        public string ban_duration;

        /// <summary>
        /// The timestamp when the ban expires (empty if permanent).
        /// </summary>
        public string banned_until;
    }
}

File: java/src/main/java/com/michitai/multiplayer/players/PlayerBanResponse.java
package com.michitai.multiplayer.players;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response containing the ban details.
 */
public class PlayerBanResponse extends ApiResponse {
    @JsonProperty("message")
    private String message;

    @JsonProperty("ban_id")
    private String banId;

    @JsonProperty("player_id")
    private int playerId;

    @JsonProperty("ban_duration")
    private String banDuration;

    @JsonProperty("ban_reason")
    private String banReason;

    @JsonProperty("banned_until")
    private String bannedUntil;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getBanId() {
        return banId;
    }

    public void setBanId(String banId) {
        this.banId = banId;
    }

    public int getPlayerId() {
        return playerId;
    }

    public void setPlayerId(int playerId) {
        this.playerId = playerId;
    }

    public String getBanDuration() {
        return banDuration;
    }

    public void setBanDuration(String banDuration) {
        this.banDuration = banDuration;
    }

    public String getBanReason() {
        return banReason;
    }

    public void setBanReason(String banReason) {
        this.banReason = banReason;
    }

    public String getBannedUntil() {
        return bannedUntil;
    }

    public void setBannedUntil(String bannedUntil) {
        this.bannedUntil = bannedUntil;
    }
}
